content/en/blog/_posts/2021-12-07-dual-stack-networking-ga.md

```diff
@@ -0,0 +1,47 @@
+
+
+## Service API updates
+
+Services were single-stack before 1.20, so using both IP families meant creating one Service per IP family. The user experience was simplified in 1.20, when Services were re-implemented to allow both IP families, meaning a single Service can handle both IPv4 and IPv6 workloads. Dual-stack load balancing is possible between services running any combination of IPv4 and IPv6.
+
+The Service API now has new fields to support dual-stack, replacing the single ipFamily field.
+* You can select your choice of IP family by setting `ipFamilyPolicy` to one of three options: SingleStack, PreferDualStack, or RequireDualStack. A service can be changed between single-stack and dual-stack (within some limits).
+* Setting `ipFamilies` to a list of families assigned allows you to set the order of families used.
+* `clusterIPs` is inclusive of the previous `clusterIP` but allows for multiple entries, so it’s no longer necessary to run duplicate services, one in each of the two IP families. Instead, you can assign cluster IP addresses in both IP families. Note that for a given pod, there is no possibility of setting multiple IP addresses in the same family.
+
+
+## Default behavior remains single-stack
+
+Starting in 1.20, the underlying networking for Kubernetes has included dual-stack whether or not a cluster was configured with the feature flag to enable dual-stack. Now that the feature is stable in 1.23, the feature flag is removed.
+
+However, Pods and Services default to single-stack unless `ipFamilyPolicy` specifies PreferDualStack or RequireDualStack. Even though dual-stack is possible, it is not mandatory to use it. Examples in the documentation show the variety possible in [dual-stack service configuration](/docs/concepts/services-networking/dual-stack/#dual-stack-service-configuration-scenarios).
+
+
+## Try dual-stack right now
+
+While upstream Kubernetes now supports [dual-stack networking](/docs/concepts/services-networking/dual-stack/) as a GA or stable feature, each provider’s support of dual-stack Kubernetes may vary. Nodes need to be provisioned with routable IPv4/IPv6 network interfaces. Pods need to be dual-stack. The [network plugin](/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/) is what assigns the IP addresses to the Pods, so the network plugin being used for the cluster needs to support dual-stack. Some Container Network Interface (CNI) plugins support dual-stack, as does kubenet.
```

Comment:
FYI kubenet has a deprecation notice https://groups.google.com/g/kubernetes-sig-network/c/nW_1gYTqW_0/m/fKkl2oNGBQAJ

Reply:
Hmm, do you think that means I should not mention it? Or should I link to something about it?

Reply:
I think we can omit `Some Container Network Interface (CNI) plugins support dual-stack, as does kubenet.` since the previous paragraph explains it ```The [network plugin](/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/) is what assigns the IP addresses to the Pods, so the network plugin being used for the cluster needs to support dual-stack. ```

Reply:
I just read Kal comments, I think I'm being too picky, this is ok for a blog post